src/components/Genre.tsx

```diff
@@ -0,0 +1,27 @@
+import { useState } from "react";
+
+interface Props {
+  genres: string[];
+  selectedGenre: string;
+  onSelect: (genre: string) => void;
+}
+export default function Search({ genres, selectedGenre, onSelect }: Props) {
+  const [currentGenre, setCurrentGenre] = useState(selectedGenre);
```

Comment:
This is the redundant state for this component. The selected genre is provided from parent via props => the state of selected ganre needs to 'live' in the parrent and update in the onSelect handler

Reply:
Moved state to parent component